src/deposit/execute_deposit_utils.cairo

```diff
@@ -80,29 +104,390 @@ struct ExecuteDepositCache {
+            token_in_price: prices.short_token_price,
+            token_out_price: prices.long_token_price,
+            amount: cache.short_token_amount,
+            price_impact_usd: precision::mul_div(
+                i128_to_u128(cache.price_impact_usd),
+                cache.short_token_usd,
+                cache.long_token_usd + cache.short_token_usd
+            )
+        };
+
+        cache.received_market_tokens += _execute_deposit(@params, @_params);
+    }
+
+    if cache.received_market_tokens < deposit.min_market_tokens {
+        DepositError::MIN_MARKET_TOKENS(cache.received_market_tokens, deposit.min_market_tokens);
+    }
+
+    market_utils::validate_market_token_balance(params.data_store, market);
+
+    (params.event_emitter)
+        .emit_deposit_executed(
+            params.key,
+            cache.long_token_amount,
+            cache.short_token_amount,
+            cache.received_market_tokens,
+        );
+
+    let event_data = EventLogData {
+        uint_items: UintItems { items: array![], array_items: array![] }, cant_be_empty: 0
+    };
+    event_data.uint_items.init_items(1);
+    event_data.uint_items.set_item(0, 'received_market_tokens', cache.received_market_tokens);
+    after_deposit_execution(params.key, deposit, event_data, params.event_emitter);
+
+    pay_execution_fee_deposit(
+        params.data_store,
+        params.event_emitter,
+        params.deposit_vault,
+        deposit.execution_fee,
+        params.starting_gas,
+        params.keeper,
+        deposit.account,
+    );
 }
 
 /// Executes a deposit.
 /// # Arguments
-/// * `params` - ExecuteDepositParams.
-/// * `_params` - _ExecuteDepositParams.
+/// * `params` - @ExecuteDepositParams.
+/// * `_params` - @_ExecuteDepositParams.
 #[inline(always)]
-fn _execute_deposit(params: ExecuteDepositParams, _params: _ExecuteDepositParams) -> u128 {
-    //TODO
-    0
+fn _execute_deposit(params: @ExecuteDepositParams, _params: @_ExecuteDepositParams) -> u128 {
+    let fees = get_swap_fees(
+        *params.data_store,
+        *_params.market.market_token,
+        *_params.amount,
+        *_params.price_impact_usd > 0,
+        *_params.ui_fee_receiver,
+    );
+
+    fee_utils::increment_claimable_fee_amount(
+        *params.data_store,
+        *params.event_emitter,
+        *_params.market.market_token,
+        *_params.token_in,
+        fees.fee_receiver_amount,
+        deposit_fee_type(),
+    );
+
+    fee_utils::increment_claimable_ui_fee_amount(
+        *params.data_store,
+        *params.event_emitter,
+        *_params.ui_fee_receiver,
+        *_params.market.market_token,
+        *_params.token_in,
+        fees.ui_fee_amount,
+        ui_deposit_fee_type(),
+    );
+
+    (*params.event_emitter)
+        .emit_swap_fees_collected(
+            *_params.market.market_token,
+            *_params.token_in,
+            *_params.token_in_price.min,
+            'deposit',
+            fees,
```

Comment:
@sparqet when building the project, I'm getting the following error here:

```
error: Variable was previously moved.
 --> execute_deposit_utils.cairo:308:13
            fees
            ^**^
note: variable was previously used here:
  --> execute_deposit_utils.cairo:308:13
            fees
            ^**^
```

I'm a bit baffled by this as this and the lines before are practically the same as [`swap_utils#L208-L233`](
https://github.com/keep-starknet-strange/satoru/blob/main/src/swap/swap_utils.cairo#L208-L233)
and [`swap_utils#L351-L365`](https://github.com/keep-starknet-strange/satoru/blob/main/src/swap/swap_utils.cairo#L351-L365). Do you have any idea where the ownership/moving error might come from?

Reply:
You need to add Copy in the derive of SwapFees struct